File: VB/WindowsApplication1/MyRibbonControl/MyRibbonViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars.Ribbon.ViewInfo

Namespace WindowsApplication1
	Public Class MyRibbonViewInfo
		Inherits RibbonViewInfo
		Public Sub New(ByVal ribbon As RibbonControl)
			MyBase.New(ribbon)

		End Sub

		Protected Overrides Function CreateGroupLayoutCalculator(ByVal pageInfo As RibbonPageGroupViewInfo) As RibbonPageGroupLayoutCalculator
			Return New MyRibbonPageGroupComplexLayoutCalculator(pageInfo)
		End Function

		Public Function CanMinimizePageGroup(ByVal group As RibbonPageGroup) As Boolean
			Dim ribbonControl As MyRibbonControl = CType(Ribbon, MyRibbonControl)
			Return ribbonControl.RaiseCanMinimizePageGroup(group, Panel)
		End Function
	End Class
End Namespace

File: CS/WindowsApplication1/MyRibbonControl/MyRibbonViewInfo.cs
using System;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars.Ribbon.ViewInfo;

namespace WindowsApplication1
{
	public class MyRibbonViewInfo : RibbonViewInfo
	{
		public MyRibbonViewInfo(RibbonControl ribbon)
			: base(ribbon)
		{
		}

		protected override RibbonPageGroupLayoutCalculator CreateGroupLayoutCalculator(RibbonPageGroupViewInfo pageInfo)
		{
			return new MyRibbonPageGroupComplexLayoutCalculator(pageInfo);
		}

		public bool CanMinimizePageGroup(RibbonPageGroup group)
		{
			MyRibbonControl ribbonControl = (MyRibbonControl)Ribbon;
			return ribbonControl.RaiseCanMinimizePageGroup(group, Panel);
		}
	}
}